org.eclipse.winery.model.tosca/src/main/java/org/eclipse/winery/model/tosca/TExtensibleElements.java

```diff
@@ -101,19 +105,59 @@ public int hashCode() {
     @NonNull
     public List<TDocumentation> getDocumentation() {
         if (documentation == null) {
-            documentation = new ArrayList<TDocumentation>();
+            documentation = new ArrayList<>();
         }
         return this.documentation;
     }
 
     @NonNull
     public List<Object> getAny() {
         if (any == null) {
-            any = new ArrayList<Object>();
+            any = new ArrayList<>();
         }
         return this.any;
     }
 
+    /**
+     * This is a special method for Winery. Winery allows to define a property definition by specifying name/type
+     * values. Instead of parsing the extensible elements returned TDefinitions, this method is a convenience method to
+     * access this information
+     *
+     * @return a WinerysPropertiesDefinition object, which includes a map of name/type-pairs denoting the associated
+     * property definitions. A default element name and namespace is added if it is not defined in the underlying XML.
+     * null if no Winery specific KV properties are defined for the given entity type
+     */
+    @XmlTransient
+    @JsonIgnore
+    public WinerysPropertiesDefinition getWinerysPropertiesDefinition() {
+        // similar implementation as org.eclipse.winery.repository.resources.entitytypes.properties.PropertiesDefinitionResource.getListFromEntityType(TEntityType)
+        WinerysPropertiesDefinition res = null;
+        for (Object o : this.getAny()) {
+            if (o instanceof WinerysPropertiesDefinition) {
+                res = (WinerysPropertiesDefinition) o;
+            }
+        }
+
+        if (res != null) {
+            // we put defaults if elementname and namespace have not been set
+            setWPDElement(res);
+            setWPDNamespace(res);
+        }
+
+        return res;
+    }
+
+    protected void setWPDElement(WinerysPropertiesDefinition res) {
+        if (res.getElementName() == null) {
+            res.setElementName("Properties");
+        }
+    }
+
+    protected void setWPDNamespace(WinerysPropertiesDefinition res) {
```

Comment:
You could actually implement it on this level since the methods are identical. To be able to use the `getTargetNamespace` method, you can check if the current object is an `instance of HasTargetNamespace` and cast it. Otherwise, you could use a default namespace.